packages/geo/src/lib/datasource/shared/datasources/wfs-datasource.ts

```diff
@@ -2,139 +2,97 @@ import olSourceVector from 'ol/source/Vector';
 
 import { OgcFilterWriter } from '../../../filter/shared/ogc-filter';
 import { OgcFilterableDataSourceOptions } from '../../../filter/shared/ogc-filter.interface';
+import { formatWFSQueryString, defaultWfsVersion, defaultFieldNameGeometry, defaultMaxFeatures, gmlRegex, jsonRegex, checkWfsParams} from './wms-wfs.utils';
 
 export class WFSDataSource extends DataSource {
   public ol: olSourceVector;
-  public ogcFilterWriter: OgcFilterWriter;
 
   constructor(
     public options: WFSDataSourceOptions,
     protected wfsService: WFSService
   ) {
     super(options);
-    this.options = this.wfsService.checkWfsOptions(options);
-    this.ogcFilterWriter = new OgcFilterWriter();
+    this.options = checkWfsParams(options);
     this.wfsService.getSourceFieldsFromWFS(this.options);
   }
 
   protected createOlSource(): olSourceVector {
+    const ogcFiltersDefaultValue = true; // default values for wfs.
+    let ogcFilters  = (this.options as OgcFilterableDataSourceOptions).ogcFilters;
     // reassignation of params to paramsWFS and url to urlWFS to have a common interface with wms-wfs datasources
     this.options.paramsWFS = this.options.params;
+    const paramsWFS = this.options.paramsWFS;
     this.options.urlWfs = this.options.url;
-    // default wfs version
-    this.options.paramsWFS.version = this.options.paramsWFS.version
-      ? this.options.paramsWFS.version
-      : '2.0.0';
-    const ogcFiltersDefaultValue = true; // default values for wfs.
-    (this.options as OgcFilterableDataSourceOptions).ogcFilters =
-      (this.options as OgcFilterableDataSourceOptions).ogcFilters === undefined
-        ? {}
-        : (this.options as OgcFilterableDataSourceOptions).ogcFilters;
-    (this.options as OgcFilterableDataSourceOptions).ogcFilters.enabled =
-      (this.options as OgcFilterableDataSourceOptions).ogcFilters.enabled ===
-      undefined
-        ? ogcFiltersDefaultValue
-        : (this.options as OgcFilterableDataSourceOptions).ogcFilters.enabled;
-    (this.options as OgcFilterableDataSourceOptions).ogcFilters.editable =
-      (this.options as OgcFilterableDataSourceOptions).ogcFilters.editable ===
-      undefined
-        ? ogcFiltersDefaultValue
-        : (this.options as OgcFilterableDataSourceOptions).ogcFilters.editable;
-
-    const baseWfsQuery = 'service=WFS&request=GetFeature';
-    // Mandatory
-    const url = this.options.urlWfs;
-    // Optional
-    const outputFormat = this.options.paramsWFS.outputFormat
-      ? 'outputFormat=' + this.options.paramsWFS.outputFormat
-      : '';
-    const wfsVersion = this.options.paramsWFS.version
-      ? 'version=' + this.options.paramsWFS.version
-      : 'version=' + '2.0.0';
-
-    let paramTypename = 'typename';
-    let paramMaxFeatures = 'maxFeatures';
-    if (
-      this.options.paramsWFS.version === '2.0.0' ||
-      !this.options.paramsWFS.version
-    ) {
-      paramTypename = 'typenames';
-      paramMaxFeatures = 'count';
```

Comment:
The next few lines modify the options. I don't think this is the right place to do that. See one of my previous comments.

Reply:
done